### ISSUE TITLE
similar to "Desktop" tab is missing? #2247 

### ISSUE THREAD

**Issue by BertJd** [Ylianst/MeshCentral#3116]
Hi,

we also have problems, that the tab sometimes is missing, when login to our MeshCentral server. We made two observations, that may be helpfull:
First: The effect often depends on the user/group and there rights! Several times, when "remote"-users lost the "Desktop" tab, administrators had their full views and could connect to the device. (Website refresh on the remote-user browser had no effects, changing their rights deactivate and activate sometimes take effects).
Second: We suppose, that the disappearence correlate to network disconnections and network issues: Actually, our devices use VSL internet connections, which are disconnected at least every 24 hours by the IP provider. The agent reconnects after the break, but right there, the tab more often is missing. It gets back most times by >clear the core<, sometimes with >upload default core< or even >soft disconnect agent<. (One time, the tab appears, but "connect" could not established. But this seems to be not often.)
Previously, the devices were connected to a dedicated internet line. The effect was much rarer, but was still there. We didn’t pay much attention to it. But the situation is now quite different.
Thus, we actually restart the agent service after each network disconnect. The effect decreases, but also there it is not completely gone - thats to little success.

perhapts you have a hint or solution, we hope,
best regards,
Bert

> info
Current Core: Aug 30 2021, 4052687721
Agent Time: 2021-09-08 12:49:00.750+02:00.
User Rights: 0xffffffff.
Platform: linux.
Capabilities: 15.
Server URL: wss://support.uiservice.net:443/agent.ashx.
OS: Garz & Fricke Yocto BSP (based on Poky) rocko-7.0-0 (rocko).
Modules: linux-dhcp, monitor-border, sysinfo, util-agentlog, wifi-scanner-windows, wifi-scanner.
Server Connection: true, State: 1.
X11 support: true.

**Comment by Ylianst** [comment]
For details on agent updating and tips and tricks, check out the [MeshCentral Agent Update video](https://www.youtube.com/watch?v=VWcDYxbSaYA).

Does this problem only occur when the remote machine is Linux? I guess one possibility is that X Window loads later than an agent on some platforms and causes the agent to send data indicating there is no X Windows where there is (but loaded later).

When the "Desktop" tab is missing, is the "Terminal" tab always present? If so, I am guessing there is a problem with the agent detecting X Windows.

**Comment by BertJd** [comment]
The agent is killed and restartet automatically after each network connection loss. Would the X Windows detection possibly fail, even if X11 is not restarted? We had hoped, that not. And isn't the fact, that the "Desktop tab" is missing in one usergroup, while it is present in another usergroup at the same time, couldn't that be an indication, that the effect is more a server-side effect than on the agent side? Would there be another criterium or how could that be verified more in detail?
Unfortunately, we only had run embedded linux devices, so far. Today, we installed an windows client. Til now, for "remote-users", "no terminal"/"no files"/"no IntelrAMT" was always set. That's also going to be chanced for testing, too. Besides we remember, that we indeed had the X Window detection problems with older versions, longer time ago. We belive, that than the info showed "X11 support: false". Currently, the "info" seams always to show "X11 support: true".

thank you,
best regards,
 Bert
